File: .config/alacritty/fonter.py
#!python
"""Import the configuration file of Alacritty Terminal, update the font size as specified, and then exit.
License: https://unlicense.org/
Author: Myxi (github.com/eeriemyxi)
Year: 2023 
"""

import sys

import yaml

USER = "eeriemyxi"


if __name__ == "__main__":
    try:
        font_size = int(sys.argv[1])
    except ValueError:
        raise Exception("That's not a valid value to put as the font.")

    with open(f"/home/{USER}/.config/alacritty/alacritty.yml", "r") as file:
        yml = yaml.safe_load(file.read())
        yml["font"]["size"] = font_size

    with open(f"/home/{USER}/.config/alacritty/alacritty.yml", "w") as file:
        file.write(yaml.dump(yml))

File: .config/alacritty/fonter.lua
local lyaml = require("lyaml")


local alacritty_config_file_read_buffer = io.open("/home/eeriemyxi/.config/alacritty/alacritty.yml", "r")

io.input(alacritty_config_file_read_buffer)
local alacritty_config_file = io.read("*all")
local yml = lyaml.load(alacritty_config_file)

yml["font"]["size"] = tonumber(arg[1])

local alacritty_config_file_write_buffer = io.open("/home/eeriemyxi/.config/alacritty/alacritty.yml", "w")
io.output(alacritty_config_file_write_buffer)
io.write(lyaml.dump({
  yml
}))
